R/dvhydrograph-data.R

```diff
@@ -1,251 +1,138 @@
+    } else {
+      readNonEstimatedTimeSeries(reportObject, seriesField, timezone=timezone, descriptionField=descriptionField, isDV=TRUE)
+    }
+  }, error=function(e) {
+    warning(paste("Returning NULL for DV Hydro Time Series: {", seriesField, "}. Error:", e))
     return(NULL)
-  }
+  })
 
-  #Get the time series type, logAxis, and approval series from the first
-  #time series that exists in the data.
-  if(!isEmptyOrBlank(stat1Timeseries)){
-    type <- stat1Timeseries[['type']]
-    logAxis <- isLogged(stat1Timeseries[['points']], stat1Timeseries[['isVolumetricFlow']], excludeZeroNegativeFlag)
-    approvalSeries <- stat1Timeseries
-  } else if(!isEmptyOrBlank(stat2Timeseries)){
-    type <- stat2Timeseries[['type']]
-    logAxis <- isLogged(stat2Timeseries[['points']], stat2Timeseries[['isVolumetricFlow']], excludeZeroNegativeFlag)
-    approvalSeries <- stat2Timeseries
-  } else if(!isEmptyOrBlank(stat3Timeseries)){
-    type <- stat3Timeseries[['type']]
-    logAxis <- isLogged(stat3Timeseries[['points']], stat3Timeseries[['isVolumetricFlow']], excludeZeroNegativeFlag)
-    approvalSeries <- stat3Timeseries
-  } else {
-    type <- ""
-    logAxis <- FALSE
-    approvalSeries <- list()
+  if(isEmptyOrBlank(timeSeries) || !anyDataExist(timeSeries[['points']])){
+    return(NULL)
   }
 
-  #Get max and min IV points
-  max_iv <- getMinMaxIV(reportObject, "MAX", timezone, type, invertedFlag)
-  min_iv <- getMinMaxIV(reportObject, "MIN", timezone, type, invertedFlag)
-
-  #If we are excluding min/max points or if we are excluding zero / negative
-  #points and the max/min vlaues are zero / negative, then replace them
-  #with labels that go on the top of the chart.
-
-  #Max Checking
-  if( (!isEmptyOrBlank(excludeMinMaxFlag) && excludeMinMaxFlag) || 
-      (!isEmptyOrBlank(excludeZeroNegativeFlag) && excludeZeroNegativeFlag && !isEmptyOrBlank(max_iv$value) && max_iv$value <= 0)){
-    max_iv_label <- max_iv
-    not_include <- c(not_include, 'max_iv')
-  } 
-
-  #Min Checking
-  if( (!isEmptyOrBlank(excludeMinMaxFlag) && excludeMinMaxFlag) 
-     || (!isEmptyOrBlank(excludeZeroNegativeFlag) && excludeZeroNegativeFlag && !isEmptyOrBlank(min_iv$value) && min_iv$value <= 0) ){
-    min_iv_label <- min_iv
-    not_include <- c(not_include, 'min_iv')
-  }
+  return(timeSeries)
+}
 
-  #Approvals
-  approvals <- readApprovalBar(approvalSeries, timezone, legend_nm=approvalSeries[['legend.name']], snapToDayBoundaries=TRUE)
+#' Parse DV Approvals
+#'
+#' @description Given a time series, read and format the approvals
+#' data from it into apprpoval bars to put on the plot.
+#' @param timeSeries the time series to get approvals for
+#' @param timezone the timezone to parse the approval times into
+parseDVApprovals <- function(timeSeries, timezone){
+  return(readApprovalBar(timeSeries, timezone, legend_nm=timeSeries[['legend.name']], snapToDayBoundaries=TRUE))
```

Comment:
separate `return` from function called, i.e. move readApprovalBar to a separate line and save to an object, then return that object. It's easier for debugging later.